account_financial_report/report/trial_balance.py

```diff
@@ -226,21 +226,37 @@ def _compute_account_amount(
                     )
         return total_amount
 
+    @api.model
+    def _prepare_total_amount(self, foreign_currency):
+        res = {
+            "credit": 0.0,
+            "debit": 0.0,
+            "balance": 0.0,
+            "initial_balance": 0.0,
+            "ending_balance": 0.0,
+        }
+        if foreign_currency:
```

Comment:
Shouldn't this be initialized always? On v14, `amount_currency` and `currency_id` is always filled in the `account.move.line`.

Reply:
Ok, changes done.

Reply:
I was just asking. What I don't want also is to have filled such columns when no different currency is set. You tell me...

Reply:
Before this PR and now, the value of the corresponding columns are filled in (I didn't want to make any "extra" changes to avoid side effects).

Taking into account `amount_currency` and `currency_id` fields is probably interesting, but I prefer not to change the current behaviour.

Reply:
But I'm not saying that. I'm asking if current behavior is correct after the previous changes you did are correct. Is foreign currency being shown in the PDF/XLSX only when it's different from the main currency?

Reply:
Ok sorry, now I understand what you are saying.

Before if the initial and final balance was shown (in the corresponding currency) but NOT the currency column, I have added a change to show this data.